magento/magento2
# Invalidate indexers in RecurringData

### Description

Calling `$indexer->reindexAll();` can take a long time with a large database.
This code is run during `php bin/magento setup:upgrade`, which is part of a
typical deployment.

This change makes the step significantly quicker, while still achieving the
same result. Minimising downtime during deployments seems sensible.

### Fixed Issues
- Fixes:  #19469
- See also:  #21235

### Manual testing scenarios
1. For best results, create a large number (eg 60k) of customers. This step is not required but will more clearly highlight the difference introduced in this change.
1. Run `php bin/magento setup:upgrade`
1. Observe (lack of) delay when running `Magento_Customer` recurring data setup task.

### Questions or comments
I could not see any existing tests covering this code.

### Contribution checklist (*)
